### PR TITLE
Adding tests and documentation for package path.

### PR DESCRIPTION
**Pull request by IsaacDunlap**
Adds new test cases, improves test coverage, adds documentation and improves code maintainability.

License: MIT
Signed-off-by: Isaac Dunlap <isaacdunlap@gmail.com>

### CODE REVIEW

File: path/path.go [hunk 1]
```diff
@@ -5,26 +5,31 @@ import (
...
 }
 
-// FromCid safely converts a cid.Cid type to a Path type
+// FromCid safely converts a cid.Cid type to a Path type.
 func FromCid(c *cid.Cid) Path {
 	return Path("/ipfs/" + c.String())
 }
 
+// Segments returns the individual segments of the Path (separated by
+// the '/' character.
```
Comment (whyrusleeping) [inline path/path.go]: missing a close paren

File: path/path.go [hunk 2]
```diff
@@ -116,15 +127,19 @@ func ParseCidToPath(txt string) (Path, error) {
 	return FromCid(c), nil
 }
 
+// IsValid returns true if the path is valid, and false otherwise.
```
Comment (whyrusleeping) [inline path/path.go]: doesnt return true or false, it returns an error if its invalid, and nil otherwise

**Comment by IsaacDunlap** [comment]
Hi,

I notice that some of the tests fail on mater but others don't. Which tests should I be worried about?

Also, how do I rerun these tests locally so I can get diags and stuff?

Isaac

**Comment by Stebalien** [comment]
> I notice that some of the tests fail on mater but others don't. Which tests should I be worried about?

All the tests should pass (usually, there are a few tests that'll fail once in a while).


> Also, how do I rerun these tests locally so I can get diags and stuff?

`make test`

Or, if you're interested in a specific test:

1. If it's a go test, go to the package in question and run `go test`.
2. If it's a sharness test, just run the appropriate shell script in `test/sharness`.

**Comment by Stebalien** [comment]
The jenkins failure may be spurious. However, the circleci failure looks real. Try running `test/sharness/t0053-dag.sh` locally.

```sh
> cd test/sharness
> make deps
> ./t0053-dag.sh
```

**Comment by Kubuxu** [comment]
I would recommend running it by: `make test/sharness/t0053-dag.sh` in rood project directory. This way you are sure dependencies are up to date.

Other way would be to run `make t0053-dag.sh` in `test/sharness` which is equivalent to the former.

**Comment by dignifiedquire** [comment]
@kubuxu did you happen to pick this up?

**Pull request closed without merging**